Reformatted codebase to follow pep8 guidelines

Monal5031 opened the pull request:
Used coala to fix all formatting issues locally.
Integrated coala with travis to raise formatting issues in PR itself from now on.

Fixes #101

Monal5031 commented:
@pombredanne @singh1114 @RajuKoushik Please review this PR the travis is now green 😄 , sorry for late fix didn't get a chance to update the branch.
I've used `coala` locally to fix the formatting issues and if this is merged the formatting errors will be shown in PR itself under the `python 3.6` tab of travis. Also coala follows `pep8` guidelines while formatting.

pombredanne commented on scanapp/migrations/0001_initial.py:
```diff
@@ -20,108 +20,152 @@ class Migration(migrations.Migration):
         migrations.CreateModel(
             name='Copyright',
             fields=[
-                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
```
Please do NOT reformat or change auto-generated migrations

Monal5031 replied:
Sure, let me try something to ignore migrations directory.

jayvdb replied:
Or improve the generator?

pombredanne replied:
@jayvdb this is the generator that is built-in Django and I am not sure it makes sense to spend any time to update this. Generated code should never be linted IMHO.

pombredanne commented on .travis.yml:
```diff
@@ -1,5 +1,9 @@
-python: 2.7
+python:
+  - 2.7
+  - 3.6
```
We do not support Python 3.6 yet at this stage since ScanCode does not support it for now

Monal5031 replied:
Yeah keeping that in mind I added these two versions, as coala will need python3+ versions. You can see in this file that I used conditional logic to run ScanCode in python2.7 and coala in python3.6

pombredanne replied:
Cannot you just run coala in the same build run? e.g. python3 is likely available in both cases.

Monal5031 replied:
I tried running it in same build but travis failed each time, here are the failed logs:
https://travis-ci.org/nexB/scancode-server/builds/288168067
https://travis-ci.org/nexB/scancode-server/builds/288178128
https://travis-ci.org/nexB/scancode-server/builds/289554875
I'll give it a try running it in the same build, if you wish.

jayvdb replied:
It is usually better to run the linters in a separate job, so that they run concurrently with the unit tests

You can use `matrix: include: ` to add an extra job for `coala-ci`.  The matrix entry can have its own `script:` etc, so you do not need to modify the main job.

pombredanne commented on .travis.yml:
```diff
@@ -10,16 +14,28 @@ addons:
+  - if [[ $TRAVIS_PYTHON_VERSION == 2.7 ]]; then 
+      pip install -r requirements.txt;
+      pip install pycodestyle;
+    else
```
I am not sure I understand why 3.6 is needed and what it does?

Monal5031 replied:
The thing is scancode works on python2.7 while coala works on python3+ versions. So to cater to both of them I added this workaround, which will run travis on 2 different versions of python, 2.7  for scancode and 3.6 for coala.

pombredanne commented on .travis.yml:
```diff
@@ -10,16 +14,28 @@ addons:
 script:
-  - python manage.py migrate
-  - python manage.py test
+  - if [[ $TRAVIS_PYTHON_VERSION == 3.6 ]]; then
```
Same as above: I am confused by the tests on the python version.

pombredanne commented on scanapp/migrations/0002_auto_20170815_2324.py:
```diff
@@ -15,6 +15,7 @@ class Migration(migrations.Migration):
         migrations.AlterField(
             model_name='scan',
             name='scan_directory',
-            field=models.CharField(blank=True, help_text='Directory in which the code to be scanned is stored', max_length=2000, null=True),
```
Same: do not reformat migrations

pombredanne commented on scanapp/models.py:
```diff
@@ -42,6 +42,7 @@ class Scan(models.Model):
     """
     Store various attributes of a scan
     """
+
```
What is this extra line about?

Monal5031 replied:
`Method definitions inside a class are surrounded by a single blank line.`
This is a point in pep8 guideline, I guess that's why coala added a new line while re-formatting.

jayvdb replied:
Ya, there should be a blank line above methods, like `__str__`.
This rule can be disabled if scancode maintainers prefer to not have a space here.

pombredanne commented on scanapp/models.py:
```diff
@@ -108,29 +110,39 @@ class License(models.Model):
     scanned_file = models.ForeignKey(ScannedFile)
     key = models.CharField(max_length=200, help_text='Key of license')
     score = models.IntegerField(help_text='Score of license')
-    short_name = models.CharField(max_length=200, help_text='Short name of the license')
```
I am not too keen on this type of strict PEP8 line wrapping for the sake of it. I consider that lines should be ~ 80ish in length. Here you are reformatting readable code in something that is no longer readable. What's the point?

Monal5031 replied:
@pombredanne If you wish to increase the number of characters allowed per line, coala has a feature for that too. We can define ` max_line_length ` according to our need. 😄

pombredanne replied:
I did not see any of the proposed line breakings that made the code more beautiful and easier to read.

Monal5031 replied:
Okay if it isn't much of a use, then the better alternative I guess is to drop the idea of reformatting. your views?

pombredanne replied:
Formatting is fine, as long as we have the right rules such as line length in the 80ish range. Skip migrations, etc...

jayvdb replied:
It looks like this repo should have `max_line_length = 100`.  There should only be a few instances of violations of that limit.
Then it will be easier to review and consider those violations.

If there are still more than 20 violations of `max_line_length = 100` , increase it to `max_line_length = 120`.
The goal of adding a `.coafile` is _not_ to fix all the problems, but instead it is to prevent future regressions making the codebase style even less unified.
After the `.coafile` is merged and being checked by CI, additional improvements can be made to style rules incrementally.

Monal5031 replied:
Okay will do it asap :+1:

pombredanne reviewed (changes requested):
Thank you!... See my comments: most of your proposed changes are about strict PEP8 line wrapping for the sake of it. I consider that lines should be ~ 80ish in length and line wrapping for the sake of PEP8 in many case destroy readability. Could you give a shot at viewing https://www.youtube.com/watch?v=wf-BqAjZb8M

Monal5031 committed:
Reformatted whole code to follow pep8 guidelines

Used coala to fix all formatting issues locally
Integrated coala with travis to point formatting issues
in PR itself from now on.

Closes https://github.com/nexB/scancode-server/issues/101

Monal5031 commented:
@pombredanne @jayvdb Please take a look now. I tried `max_line_length=100` but it still had many violations, so updated with `120`.

pombredanne commented:
@Monal5031 this looks decent now alright .... but since this is touching several files, you now have a responsibility I merge this now: You will need to rebase and/or resolve any conflicts in the pending PRs in https://github.com/nexB/scancode-server/pulls ? Do you think you could do this? that should not be a huge amount of work. I can even bring in these PR  branches in this repo if this helps.

Monal5031 commented:
@pombredanne Okay sure, I checked this change won't cause much merge conflicts though we'll be needing rebases over all PRs. Also, some of them might fail Travis as they will also need to run coala. I can fix this repository if it breaks 😄

pombredanne commented:
@Monal5031 I take your word for it that you fix the mess that may come! So please start working on it now :P

Monal5031 commented:
Okay @pombredanne will head onto them as soon as this is merged 😃

pombredanne commented:
You will also need to adjust your commit comment style to be imperative in the future.

Monal5031 commented:
Sure 👍 , working on the old PRs
